File: core/src/main/java/eu/xap3y/xagui/interfaces/listeners/GuiPageSwitchInterface.java
package eu.xap3y.xagui.interfaces.listeners;

import eu.xap3y.xagui.models.GuiPageSwitchModel;
import org.jetbrains.annotations.NotNull;

public interface GuiPageSwitchInterface {
    void onPageSwitch(@NotNull GuiPageSwitchModel event);
}

File: src/main/kotlin/eu/xap3y/xagui/interfaces/GuiPageSwitchInterface.kt
package eu.xap3y.xagui.interfaces

import eu.xap3y.xagui.models.GuiPageSwitchModel

/**
 * Interface for page switch event listeners
 *
 * @see GuiInterface
 */
interface GuiPageSwitchInterface {

    /**
     * Called when the page is switched
     * @param data The data of the switch
     */
    fun onPageSwitch(data: GuiPageSwitchModel)
}